lib/dolly/document.rb

```diff
@@ -149,6 +151,20 @@ def self.property *ary
       end
     end
 
+    class << self
+      def mango_scope scope_name, scope
+        self.mango_scopes ||= {}
+        name = scope_name.to_sym
+        self.mango_scopes[name] = lambda { |query_object, args| Dolly::Scope.new(query_object, scope, args)}
+
+        (class << self; self end).instance_eval do
```

Comment:
@javierg sure.  This is so that the instance eval doesn't occur on the `Dolly::Document` singleton class, but instead the subclass in which the scope is defined, which is an instance of Class 

This small snippet is incredibly expressive.  

    [3] pry(main)> Dolly::Document.instance_of?(Class)
    => true

This demonstrates that class definitions themselves are just instances of `Class`,thus can be sent the message `instance_eval`